prow/private_deck.md

```diff
@@ -0,0 +1,188 @@
+   ```
+
+and then add two External Secrets like so:
+
+```
+apiVersion: kubernetes-client.io/v1
+kind: ExternalSecret
+metadata:
+  name: oauth-config
+  namespace: default
+spec:
+  backendType: gcpSecretsManager
+  projectId: <PROJECT>
+  data:
+  - key: <my-gsm-secret-name>
+    name: secret
+    version: latest
+---
+apiVersion: kubernetes-client.io/v1
+kind: ExternalSecret
+metadata:
+  name: oauth
+  namespace: default
+spec:
+  backendType: gcpSecretsManager
+  projectId: PROJECT
+  data:
+  - key: <my-gsm-secret-name>
+    name: clientID
+    version: latest
+    property: client_id
+  - key: <my-gsm-secret-name>
+    name: clientSecret
+    version: latest
+    property: client_secret
+  - key: <my-gsm-secret-name>
+    name: cookieSecret
+    version: latest
+    property: cookie_secret
+```
+
+## 4 Create the new Deck Deployment
+
+When creating the new Deck Deployment, make sure to update the following fields:
+
+- Update Service Account on new Deployment
+- Update TenantID on new Deployment
+    - Add `- --tenant-id=NEW_ID` under args in the Deck deployment spec
+- (Optional) Add a volume mount for the oauth app and update the oauth-config 
+    - Here is an example of oauth2-proxy being used with github account validation. The oauth and oauth-config secrets are made in step 3.
+
+    ```
+    volumeMounts:
+    ...
+    - name: oauth2-proxy
+        image: quay.io/oauth2-proxy/oauth2-proxy
+        ports:
+        - containerPort: 4180
+          protocol: TCP
+        args:
+        - --provider=github
+        - --github-org=ORG
+        - --github-team=TEAM
+        - --http-address=0.0.0.0:4180
+        - --upstream=http://localhost:8080
+        - --cookie-domain=DOMAIN
+        - --cookie-name=COOKIE NAME (can be anything)
+        - --cookie-samesite=none
+        - --cookie-expire=23h
+        - --email-domain=*
+        livenessProbe:
+          httpGet:
+            path: /ping
+            port: 4180
+          initialDelaySeconds: 3
+          periodSeconds: 3
+        readinessProbe:
+          httpGet:
+            path: /ping
+            port: 4180
+          initialDelaySeconds: 3
+          periodSeconds: 3
+        env:
+        - name: OAUTH2_PROXY_CLIENT_ID
+          valueFrom:
+            secretKeyRef:
+              name: oauth
+              key: clientID
+        - name: OAUTH2_PROXY_CLIENT_SECRET
+          valueFrom:
+            secretKeyRef:
+              name: oauth
+              key: clientSecret
+        - name: OAUTH2_PROXY_COOKIE_SECRET
+          valueFrom:
+            secretKeyRef:
+              name: oauth
+              key: cookieSecret
+        - name: OAUTH2_PROXY_REDIRECT_URL
+          value: https://prow.infra.cft.dev/oauth2/callback
+    ```
+    
+    ```
+    volumes:
+    ...
+    - name: oauth-config
+        secret:
+          secretName: oauth-config
+    ```
+
+## 5 Use the New Deployment
+
+In order to use the new Deployment you will need to:
+
+1. Make a new static IP
```

Comment:
Is this TODO? I feel this could be expanded to be with more details

Reply:
Done